### PR TITLE
[DPTOOLS-2832] override execution_timeout if task timeout is provided

### PR DESCRIPTION
**Pull request by youngyjd**
If timeout is provided, override execution_timeout

### CODE REVIEW

**Comment by youngyjd** [comment]
@milton0825 @feng-tao @astahlman

File: airflow/models.py [hunk 1]
```diff
@@ -1407,11 +1407,15 @@ def signal_handler(signum, frame):
 
                 # If a timeout is specified for the task, make it fail
                 # if it goes beyond
+                task_timeout = task_copy.execution_timeout
+                if hasattr(task_copy, 'timeout') and task_copy.timeout:
```
Comment (feng-tao) [inline airflow/models.py:1411]: isn't just `if hasattr(task_copy, 'timeout')` enough?

Comment (milton0825) [reply airflow/models.py:1411]: is `task_copy.timeout` default to None for `BaseSensor`?

Comment (youngyjd) [reply airflow/models.py:1411]: Sensor has timeout parameter, but if can be none which means we should use execution_timeout instead

Comment (youngyjd) [reply airflow/models.py:1411]: @milton0825 No. I have a PR out to change LyftExternalTaskSensor https://github.com/lyft/python-lyft-etl/pull/226

Comment (feng-tao) [reply airflow/models.py:1411]: +1

Comment (youngyjd) [reply airflow/models.py:1411]: Actually, maybe just changing LyftExternalTaskSensor will be enough. Holding off this PR.

**Pull request closed without merging**